### ISSUE TITLE
Blit a standard control in a HTML element of the shown webView 

### ISSUE THREAD

**Issue by rexxitall** [twinbasic/twinbasic#1314]
Would help game developers a lot cause they could now style the UI without the need of custom controls like they want and can blit a view to the game screen into a part of the html page shown as GUI. Might look a bit strange on first sight but it would be very handy also while porting from or to andoid / web. In my case i have a need of custom controls and they are faster done in HTML and CSS as with OCX. And you has all the benfits from that world which is hard to implement as ocx. And i do not even talk about all the controls they offer already in HTML JS and CSS. Which leads also to the prvious issue

**Comment by wqweto** [comment]
Btw, there are [discussions section](https://github.com/twinbasic/twinbasic/discussions) open in this repo for the exact purpose of discussing things.

**Comment by rexxitall** [comment]
I do not think thats a discussion theme. I would like to have this feature. You may discuss it how to implement and which kind of syntax sugar. Its also not a that high hanger, most of the needed things should be already there. To get the current position and size of a thml element is a few lines JS and then the rest is just to move the control to that size and position and route some events while also avoiding flickering. At least i think in my simple childish mind :) And for the opposite i might remeber tehere is already something wehre you can split a webpage in rectangles and move the content as well... At least i think that the code i saw was for that purpose :) But i might mislead. In fact both ways will improve rapid prototyping, save a lot of time, enable quick mockups, Get the HTML designer on board. Now you can outsource gui dev to them if you want. It has a lot of benfits if it would be working. In my case it would save a lot of time :) If you are in need of a special control - i am thinking for example of [Oli Folkerds Tabulator](https://github.com/olifolkerd/tabulator) How much time it would take to port just this hell of a grid table to native TB ?

**Comment by wqweto** [comment]
We'll have to leave to Wayne to decide whether to move this issue to discussions but some extra helpers for WebView can be built outside of main language development as a proof of concept.

You can even show some minimal code which does what you think is possible in a few lines of JS, route the events, etc. seems like you have a plan how to implement this.

**Comment by rexxitall** [comment]
I just can imagine how it is implement already. Wyne has as far i know a internal websock connection to the webview control. Otherwise he could not use them. So by this you will get by a simple request  the current position of the hatml element by id or whatever JS has to offer to select a element. Via a bit more JS you get the position in the page as well. That can be recalculated to get the visible area on the form to display the control in front (topmost) of the webview. If you make the control, you can also by this way add the needed event handler via js to the HTML element. They will be fired inside the js first - like resizing forced by the user (should rather be easy) or scrolling (not sure if that is needed but might be tricky) . This events can be routed also via websock to the control. Might be a more clever faster way then the websock but in principle it will work. I just do not know if its fast enough that way to avoid flickering. Might need some logik which thing is to be  draw / update first. The webview or the control in front of.  It is no urgend thing. More something to keep in the back of the head to implement some days if the compiler is mature enough. I have here such a hybrid solution up and running. I control Autocad via websockets. So i have a websock server, written a websock connection via vb.net which offers the websock connection as com object to VBA. Sure a shot back trough the neck into the eye. But works fine and fast :)  In my case its also database driven so i can display parts of the things in Autocad Panels. Autocad is able to display Websites inside a user created Panel and can also communicate with JS to that. I just want to have the ability to work sometimes with the same thing without starting autocad first :) So i use a normal webserver in my case. But as far as i know also there is already a webserver in TB as well as the websock connection :) Which was the most tricky part on that stunt :) Benefit for me. I get rid of all the control dll hell. I have a standart way to add whatever special control in whatever specual view i may need. I open the whole JS/CGI/PHP/HTML/CSS World to my program. While using most time VBA. Just not for the GUI :) Ther was no usable Code editor for VBA64 so i use Codemonkey, There was no really good data Grid i use tablator. There was no nice docking control in VB7 i adopt something from Github. They all shoot usually just the results of the changes to VBA via websock. The way back works same. On special vba events i call the websock and those datas is rerouted to the web interface to update the html values.

**Comment by wqweto** [comment]
No, you have everything mixed up. One thing is WebView control, completely different thing is TB IDE where you might find WebSockets for IPC and everything else which is not available to be used in your projects. The part written in C++ will be of no use to us anyway.

IMO this quickly already turned into a brain f*rt which should better be moved to discussions.
